op-batcher/batcher/tx_data.go

```diff
@@ -0,0 +1,52 @@
+package batcher
+
+import (
+	"fmt"
+)
+
+// txData represents the data for a single transaction.
+//
+// Note: The batcher currently sends exactly one frame per transaction. This
+// might change in the future to allow for multiple frames from possibly
+// different channels.
+type txData struct {
+	frame frameData
+}
+
+// ID returns the id for this transaction data. It can be used as a map key.
+func (td *txData) ID() txID {
+	return td.frame.id
+}
+
+// Bytes returns the transaction data. It's a version byte (0) followed by the
+// concatenated frames for this transaction.
+func (td *txData) Bytes() []byte {
+	return append([]byte{0}, td.frame.data...)
```

Comment:
do we want to use `derive.DerivationVersion0` or even a local constant?

Reply:
Yes we should, changed! I don't like magic numbers myself 🪄